releases/cosmwasm/2025-02-XRPL-EVM-GMP-v6.0.4.md

```diff
@@ -0,0 +1,353 @@
+  -t "Register Multisig Prover for $CHAINS" \
+  -d "Register Multisig Prover address for $CHAINS at Coordinator contract" \
+  --runAs $RUN_AS_ACCOUNT \
+  --deposit $DEPOSIT_VALUE \
+  --msg "{
+    \"register_prover_contract\": {
+      \"chain_name\": "$CHAINS",
+      \"new_prover_addr\": \"$MULTISIG_PROVER\"
+    }
+  }"
+```
+
+9. Authorize `$CHAINS` Multisig prover on Multisig
+
+```bash
+node cosmwasm/submit-proposal.js execute \
+  -c Multisig \
+  -t "Authorize Multisig Prover for $CHAINS" \
+  -d "Authorize Multisig Prover address for $CHAINS at Multisig contract" \
+  --runAs $RUN_AS_ACCOUNT \
+  --deposit $DEPOSIT_VALUE \
+  --msg "{
+    \"authorize_callers\": {
+      \"contracts\": {
+        \"$MULTISIG_PROVER\": "$CHAINS"
+      }
+    }
+  }"
+```
+
+```bash
+axelard q wasm contract-state smart <multisig-addr> '{"is_caller_authorized": {"contract_address": "<chain-multisig-prover-addr>", "chain_name": "$CHAINS"}}' --output json | jq .
+# Result should look like:
+{
+  "data": true
+}
+```
+
+10. Create reward pool for voting verifier
+
+#### Rewards
+
+| Network              | `epoch_duration` | `participation_threshold` | `rewards_per_epoch` |
+| -------------------- | ---------------- | ------------------------- | ------------------- |
+| **Devnet-amplifier** | `100`            | `["7", "10"]`             | `100`               |
+| **Stagenet**         | `600`            | `["7", "10"]`             | `100`               |
+| **Testnet**          | `600`            | `["7", "10"]`             | `100`               |
+| **Mainnet**          | `14845`          | `["8", "10"]`             | `TBD`               |
+
+```bash
+node cosmwasm/submit-proposal.js execute \
+  -c Rewards \
+  -t "Create pool for $CHAINS in $CHAINS voting verifier" \
+  -d "Create pool for $CHAINS in $CHAINS voting verifier" \
+  --runAs $RUN_AS_ACCOUNT \
+  --deposit $DEPOSIT_VALUE \
+  --msg "{
+    \"create_pool\": {
+      \"params\": {
+        \"epoch_duration\": [epoch duration],
+        \"participation_threshold\": [participation threshold],
+        \"rewards_per_epoch\": [rewards per epoch]
+      },
+      \"pool_id\": {
+        \"chain_name\": "$CHAINS",
+        \"contract\": \"$VOTING_VERIFIER\"
+      }
+    }
+  }"
+```
+
+11. Create reward pool for multisig
+
+```bash
+node cosmwasm/submit-proposal.js execute \
+  -c Rewards \
+  -t "Create pool for $CHAINS in axelar multisig" \
+  -d "Create pool for $CHAINS in axelar multisig" \
+  --runAs $RUN_AS_ACCOUNT \
+  --deposit $DEPOSIT_VALUE \
+  --msg "{
+    \"create_pool\": {
+      \"params\": {
+        \"epoch_duration\": [epoch duration],
+        \"participation_threshold\": [participation threshold],
+        \"rewards_per_epoch\": [rewards per epoch]
+      },
+      \"pool_id\": {
+        \"chain_name\": "$CHAINS",
+        \"contract\": \"$MULTISIG\"
+      }
+    }
+  }"
+```
+
+12. Add funds to reward pools from a wallet containing the reward funds `$REWARD_AMOUNT`
```

Comment:
for future, we have a script for funding rewards in the tools repo that automates this enough that you don't have to review it again for each deployment. @RiceAndMeet also has a PR to automate reward funding on testnet and rewards distribution too perhaps. Can resurrect that PR on the Infra repo.